src/_pytest/assertion/util.py

```diff
@@ -354,7 +354,7 @@ def _compare_eq_iterable(
 def _compare_eq_sequence(
     left: Sequence[Any],
     right: Sequence[Any],
-    highlighter: _HighlightFunc,
+    highlighter: _HighlightFunc | Callable[[str], str],
```

Comment:
This should not really be needed, given `_HighlightFunc` is compatible with `Callable[[str], str]`:

```python
class _HighlightFunc(Protocol):
    def __call__(self, source: str, lexer: Literal["diff", "python"] = "python") -> str:
        """Apply highlighting to the given source."""
```